content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/_index.md

```diff
@@ -0,0 +1,335 @@
+The artifacts are the "plain old data structures" (PODs) that are passed to and from the runtime. These artifacts come in two flavors: *static* artifacts, and *dynamic* artifacts. These artifacts are PODs to allow them to be passed to and from the runtime without the producers and consumers of these artifacts needing to be aware of how the runtime itself models the behavior of these artifacts. These artifacts are then used to instantiated more complex types that are used by directly by the runtime via the various artifact *factories*.
+
+### Static Artifacts
+
+Static artifacts are generated by the TIAF CMake scripts to model O3DE's build tree in memory. These artifacts are generated only once upon running CMake and persist between invocations of the runtime. These artifacts contain the information about each build target and its source file mappings, as well as information about whether each build target is a production target (build targets that are used to build the O3DE product itself) and test targets (build targets that contain the tests to be run for the production targets they cover).
+
+### Dynamic Artifacts
+
+Dynamic artifacts are artifacts that are produced by the runtime for a each invocation and may or may not persist between invocations. Such artifacts model the test results and coverage of test runs, having a near one to one mapping with the language and framework specific test result and coverage output of the test targets.
+
+### Artifact Factories
+
+The artifact factories take these raw POD artifacts and produce the types used internally by the runtime. The test target descriptor factory in particular consumes multiple artifacts per instantiated type due to the limitations of the ability to scrape information about the build tree at the CMake level.
+
+## Test runner stack
+
+Below is a diagram that demonstrates the test runner stack. It's something between a class diagram and an omnidirectional data flow diagram but so it's best not to take the diagram too literally. Rather, you can use it as the jumping off point for a given test runner and how you will need to drill down in the code to reach the constituent classes that make up that test runner.
+
+```mermaid
+flowchart TD
+    NativeShardedRegularTestRunner --> NativeRegularTestRunner
+    NativeShardedInstrumentedTestRunner --> NativeInstrumentedTestRunner
+    NativeRegularTestRunner --> TestRunner
+    NativeInstrumentedTestRunner --> TestRunnerWithCoverage
+    PythonInstrumentedTestRunner --> TestRunnerWithCoverage
+    TestRunner --> TestRunnerBase
+    TestRunnerWithCoverage --> TestRunnerBase
+    TestRunnerBase --> TestJobRunner
+    NativeTestEnumerator --> TestEnumerator
+    TestEnumerator --> TestJobRunner
+    TestJobRunner --> JobRunner
+    JobRunner --> ProcessShecduler
+    ProcessShecduler --> OS
+    style OS fill:#fff
+```
+
+The following is a brief overview of each part of the stack (in inverse depth order) along with its function within the stack.
```

Comment:
suggestion: Rather than 'inverse depth' say 'top-down' or 'bottom-up' to be a little clearer which you mean?

Reply:
Fixed 👍